src/pages/Flavors.js

```diff
@@ -1,36 +1,198 @@
+import { getStash, isLoaded } from 'selectors/flavor';
+import { getAllFlavors, getFlavorsPager } from 'selectors/flavors';
 
-import FlavorData from 'data/flavors';
+export class Flavors extends Component {
+  static propTypes = {
+    actions: PropTypes.object.isRequired,
+    collection: PropTypes.array.isRequired,
+    loggedIn: PropTypes.bool.isRequired,
+    pager: PropTypes.object.isRequired,
+    pagerNavigation: PropTypes.node.isRequired,
+    stash: PropTypes.array,
+    stashLoaded: PropTypes.bool.isRequired
+  };
+
+  constructor(props) {
+    super(props);
+
+    this.state = { stashControl: false };
+
+    const { actions, loggedIn, stashLoaded } = this.props;
+
+    if (loggedIn) {
+      if (!stashLoaded) {
+        actions.requestStash();
+      }
+      this.handleStashControl = this.stashController.bind(this);
+      this.handleAddToStash = this.addToStash.bind(this);
+      this.handleRemoveFromStash = this.removeFromStash.bind(this);
+    }
+  }
+
+  stashController(event) {
+    const { stash } = this.props;
+    const { holdings: stashMap } = this.state;
+
+    if (!stashMap) {
+      const holdings = [];
+
+      stash.map(flavor => {
+        holdings[flavor.flavorId] = true;
+      });
+      this.setState({ holdings });
+    }
+    const {
+      target: { checked, name }
+    } = event;
+
+    this.setState({
+      [name]: checked
+    });
+  }
+
+  addToStash(id) {
+    const { actions } = this.props;
+
+    actions.addStash({ id });
+
+    const { holdings } = this.state;
+
+    holdings[id] = true;
+
+    this.setState({ holdings });
+  }
+
+  removeFromStash(id) {
+    const { actions } = this.props;
+
+    actions.removeStash({ id });
+
+    const { holdings } = this.state;
+
+    holdings[id] = false;
+
+    this.setState({ holdings });
+  }
+
+  inStashIcon(id) {
+    return (
+      <ToggleButton
+        initialValue={true}
+        onClick={e => this.handleRemoveFromStash(id, e)}
+        title="Remove from Stash"
+        variant="check"
+      />
+    );
+  }
+
+  noStashIcon(id) {
+    return (
+      <ToggleButton
+        onClick={e => this.handleAddToStash(id, e)}
+        title="Add to Stash"
+        variant="check"
+      />
+    );
+  }
 
-export default class Flavors extends Component {
   render() {
+    const { collection, loggedIn, pager, pagerNavigation } = this.props;
+    const { holdings, stashControl } = this.state;
+
     return (
       <Container>
         <Helmet title="Flavors" />
-        <Table striped hover>
+        <Container fluid={true}>
+          <Row className="text-center">
+            <Col>
+              <h1>Flavors</h1>
+            </Col>
+          </Row>
+          <Row>
+            <Col className="text-center">
+              {loggedIn ? (
+                <Form>
+                  <Form.Check
+                    name="stashControl"
+                    type="checkbox"
+                    id="flavorStash"
+                    label="&nbsp;Enable Flavor Stash"
```

Comment:
Maybe outside the scope of this PR, but we shouldn't have to prepend &nbsp; in here to get the appropriate padding on this control - @JosefBud any ideas?

Reply:
I'm at work and unable to actually view these changes in a build, so forgive me if I'm speaking out of ignorance, but shouldn't we be using the new <ToggleButton /> component rather than a checkbox if it's outside of a form?

Reply:
Yeah, good point, can you think of an icon pairing for show/hide? I know I need to add an expand/contract for the Recipe Editor.